src/js_parser.zig

```diff
@@ -3550,6 +3551,53 @@ pub const Parser = struct {
                 uses_dirname = false;
                 uses_filename = false;
             }
+        } else if (p.options.output_format == options.Format.esm) {
+            if (uses_dirname or uses_filename) {
+                const count = @as(usize, @intFromBool(uses_dirname)) + @as(usize, @intFromBool(uses_filename));
+                var declared_symbols = DeclaredSymbol.List.initCapacity(p.allocator, count) catch unreachable;
```

Comment:
tiny nit, can all newly added `catch unreachable` on memory allocation failures be rewritten to `catch bun.outOfMemory()`